src/Kestrel.Core/Internal/Infrastructure/HttpUtilities.cs

```diff
@@ -59,7 +59,19 @@ private static unsafe ulong GetAsciiStringAsLong(string str)
             }
         }
 
-        private static unsafe uint GetAsciiStringAsInt(string str)
+        public static unsafe ushort GetAsciiStringAsShort(string str)
```

Comment:
Since you removed the optimized comparisons, does this file still need to change?